src/content/developers/docs/scaling/optimistic-rollups/index.md

```diff
@@ -1,46 +1,254 @@
+
+As mentioned, the operator of an optimistic rollup bundles off-chain transactions into a batch and sends it to Ethereum for notarization. This process involves compressing transaction-related data and publishing it on Ethereum as `calldata`. 
+
+Calldata is a non-modifiable, non-persistent area in a smart contract that behaves like [memory](/developers/docs/smart-contracts/anatomy/#memory). While calldata persists on-chain as part of the blockchain's [history logs](https://docs.soliditylang.org/en/latest/introduction-to-smart-contracts.html?highlight=memory#logs), it is not stored as a part of Ethereum's state. Because calldata does not touch any part of Ethereum's state, it is cheaper to use compared to `memory` or `storage`.
+
+The `calldata` keyword is also used in Solidity to pass arguments to a smart contract function at execution time. Calldata identifies the function being called during a transaction and holds inputs to the function in the form of an arbitrary sequence of bytes.
+
+In the context of optimistic rollups, `calldata` is used to send compressed transaction data to the on-chain contract. The rollup operator adds a new batch by calling the required function in the rollup contract and passing the compressed data as function arguments. Using `calldata` reduces user fees since most costs that rollups incur come from storing data on-chain.
+
+Here is [an example](https://etherscan.io/tx/0x9102bfce17c58b5fc1c974c24b6bb7a924fb5fbd7c4cd2f675911c27422a5591) of a rollup batch submission to show how this concept works. The sequencer invoked the `appendSequencerBatch()` method and passed the compressed transaction data as inputs using `calldata`. 
+
+### State commitments {#state-commitments} 
+
+At any point in time, the optimistic rollup’s state—accounts, balances, contract code, etc—is organized as a [Merkle tree](/whitepaper/#merkle-trees) called a “state tree”. The root of this Merkle tree, referencing the rollup’s latest state, is hashed and stored in the rollup contract. Every state transition on the chain produces a new rollup state, which an operator commits to by computing a new state root.
+
+The operator is required to submit both old state roots and new state roots when posting batches. If the old state root matches the existing state root in the on-chain contract, the latter is discarded and replaced with the new state root. 
+
+The rollup operator is also required to commit a Merkle root for the transaction batch itself. This allows anyone to prove the inclusion of a transaction in the batch (on L1) by presenting a [Merkle proof](/developers/tutorials/merkle-proofs-for-offline-data-integrity/). 
+
+State commitments, especially state roots, are neccessary for detecting fraud in optimistic rollups. They also make it easier to reverse an invalid block; the rollup contract only has to revert to the last valid block's state root to declare it the canonical block. 
```

Comment:
But the docs say the state root for the invalid block (and any others that built on it) will get deleted from the State Commitment Chain. Isn't that the same as reversing the block (even if the transactions are re-executed)?

Reply:
In Optimism the block is just a place where you put some background information that could be required to execute the transactions, followed by the transactions themselves.

Reply:
Okay, I had to go through both the Paradigm article and Arbitrum docs again, but I think I can clarify my point now:

In both Optimism and Arbitrum, the invalid batch is rejected, but the transactions themselves are still available to be re-executed and applied to the rollup's state (by the node posting the valid assertion).  I don't think invalid transactions are "erased" in Arbitrum; for a batch to be invalid, it means someone else has posted another batch that correctly executes those transactions (they have to post correct assertions before challenging). If the challenger's block/state root is valid, it becomes the new state root in the on-chain contract--but the initial state root (proposed by the malicious node) has to be deleted first. 

Here's a quote from the Paradigm article for context: 

> Finally, [finalizeFraudVerification](https://github.com/ethereum-optimism/contracts-v2/blob/0ad4dcfdef11ef87e278a8159de8414c8e329ba1/contracts/optimistic-ethereum/OVM/verification/OVM_FraudVerifier.sol#L147) is called on the Fraud Verifier contract which checks if the state transitioner is complete and if yes, it calls [deleteStateBatch](https://github.com/ethereum-optimism/contracts-v2/blob/f6069a881fbf6c35687a1676a73c67596b3ef4f9/contracts/optimistic-ethereum/OVM/chain/OVM_StateCommitmentChain.sol#L86) which proceeds to delete all state root batches after (including) the disputed transaction from the SCC. **The CTC remains unchanged, so that the original transactions are re-executed in the same order**.

Reply:
Sorry, I'm on a trip - I'll check on Wednesday, OK?

Reply:
OK, I checked it with our developers. In Optimism nobody else posts a new batch, the batch itself is still valid. The only thing that changes is the state root.

Reply:
Changed the text to read: 

> State commitments, especially state roots, are necessary for proving the correctness of state changes in an optimistic rollup. The rollup contract accepts new state roots from operators immediately they are posted, but can later delete invalid state roots to restore the rollup to its correct state. 

I think it explains how rollups operate "optimistically" and shows the core objective of posting and storing state commitments: verifying the rollup's canonical state. Let me know your thoughts on it.

Reply:
Approved